### DISCUSSION TITLE
Sync server with external database?

### DISCUSSION THREAD

**Discussion by dbr** [taskchampion/taskchampion#154] [Ideas]
I had briefly thought "could we make the sync-server require an external database"

- for: there are plenty of good KV stores, e.g Redis
- for: can delegate worrying about performance to them
- against: makes setup a bit harder
- against: implementing similar storage to client (e.g SQLite) should be quite easy, and we can add server backends later as needed

**Comment by djmitche** [comment]
Eventually, the sync server should support an external DB.  I think using SQLite is a good step on the way, especially if done with forethought, as porting to another DB will entail only making some minor changes to the SQL queries involved.

But, this is probably not something we need to do soon.  An implementation that uses sqlite will do just fine for one person, or a bunch of friends, sharing a sync server.

**Reply by djmitche** [reply]
I think we'll switch to etebase..
